### ISSUE TITLE
build.js not up to date?

### ISSUE THREAD

**Issue by DolphinIQ** [pailhead/three-instanced-mesh#18]
Hello
First of all, thanks for a great series on instancing. I havent had the time to understand all of the low level stuff, but I tried your abstract class. It looks great, but Im running into some problems when implementing it, so I have a few questions.
Is it possible to use it without npm? The page I have is just a standard html without NodeJS. I’ve tried the InstancedMesh.js file from your Interactive Cubes Demo, but the console immidiately threw two errors in my face:

    THREE.InstancedBufferAttribute now expects 'normalized' in position 3 - which I patched up quickly, by just adding an extra false argument in the InstancedMesh.js
    THREE.WebGLProgram: shader error: 0 35715 false gl.getProgramInfoLog invalid shaders ERROR: 0:238: 'uvTransform' : undeclared identifier - is a little more troublesome. I’ve tried doing some shader editing with .replace() , but had little success.

Im using THREE r101 for this project, although I might switch to 104 later. Is the build not updated?

**Comment by barnabasbartha** [comment]
I would use this package too, but it looks as if didn't work with r104 yet.

**Comment by pailhead** [comment]
I kinda snoozed on this package, will revisit it soon :(

**Comment by trusktr** [comment]
I made a pen, not working with r110, the mesh doesn't render: https://codepen.io/trusktr/pen/xxxQPxd?editors=0010

No errors in console though.

**Comment by trusktr** [comment]
What's the last version of Three.js that it is designed to work with? Does the version of three-instanced-mesh correspond to the version of Three.js?

I tried the same pen with Three r96, but no luck: https://codepen.io/trusktr/pen/yLLQPeY (for reference uncomment the `mesh` and you should see that).

**Comment by barnabasbartha** [comment]
I got this working a few weeks ago, I think it will work with r110.  See #17 #19. I even added support for multi material meshes, so maybe I can make a PR if someone is interested.

**Comment by trusktr** [comment]
Sweet. Do you mind updating one of my pens to show it working? They are very simple, I don't get why they don't work. EDIT: I tried using Three r86, but still no luck. I must be missing something simple.

**Comment by barnabasbartha** [comment]
In this case I don't think it is a problem with the package but your code. I see your box and noticed you rotate the instanced mesh itself. You should not do that but use the setQuaternionAt according to the docs.

**Comment by trusktr** [comment]
@BarthaBRW Yes, but I figured I could also rotate the whole collection of instances, just like any regular object. It does extend from `Object3D`. But even without rotation, I'd expect to at least see a box on the screen.

**Comment by trusktr** [comment]
The README says

> Instancing attributes are set taking these flags into consideration (`instancePosition`, `instanceQuaternion` and `instanceScale` are always created, `instanceColor` depends on the flag). Their arrays are **not instantiated**. _The idea behind this is noble, why do any work you are going to make reduntant right away. Scale on the other hand can likely be just 1,1,1, maybe an optional method to "init" the mesh should be provided? Please give feedback_

In particular, it says

> Their arrays are **not instantiated**.

Do I need to instantiate them? Is that what I'm missing?

**Comment by trusktr** [comment]
Ok! Well, I got it to work. It is required that you set initial values for position, scale, quaternion, and color if color is enabled, for each instance!

So I have code like this in my project now:

```js
        const v = new Vector3()
        const s = new Vector3(1, 1, 1)
        const q = new Quaternion()

        const instanced = new InstancedMesh(geom, mat, this.__numberOfModelInstances, true, false, true)
        for (let i = 0; i < this.__numberOfModelInstances; i += 1) {
          instanced.setPositionAt(i, v)
          instanced.setScaleAt(i, s)
          instanced.setQuaternionAt(i, q)
        }
```

And here's the above pen working now: https://codepen.io/trusktr/pen/wvvQmRB

@pailhead This wasn't intuitive. IMO it'd be better for this to just work by default.

So, now I'm up and running on r110.
